src/library/services/httpServices/http-data-provider-service.ts

```diff
@@ -0,0 +1,70 @@
+import {Injectable} from '@angular/core';
+import {Book} from "../../model/book";
+import {DataProvider, SearchCriteria} from "../data-provider.service";
+import {HttpClient} from "@angular/common/http";
+import {map} from "rxjs/operators";
+import {Observable} from "rxjs";
+import {environment} from "../../../environments/environment";
+
+type BookPredicate = (book: Book) => boolean;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HttpDataProvider extends DataProvider {
+  private books: Array<Book>;
+
+  constructor(private httpClient: HttpClient) {
+    super();
+  }
+
+  public findBooks(searchCriteria: SearchCriteria): Observable<Array<Book>> {
+    let body = searchCriteria || {};
+    if (searchCriteria) {
+      return this.httpClient.post<Array<Object>>(`${environment.URL}/books/search`, body)
+        .pipe(
+          map(httpResponse => {
+            console.log(httpResponse)
+            return httpResponse.map(obj => this.mapBook(obj));
+          })
+        )
+    } else {
+      return this.httpClient.get<Array<Object>>(`${environment.URL}/books`)
+        .pipe(
+          map(httpResponse => httpResponse.map(obj => this.mapBook(obj)))
+        )
+    }
+  }
+
+  public addBook(book: Book): Observable<Book>  {
+    return this.httpClient.post<Book>(`${environment.URL}/books`, book);
+  }
+
+  public removeBook(book: Book): Observable<any> {
+    return  this.httpClient.delete(`${environment.URL}/books/${book.getId()}`)
+  }
+
+  public updateBook(book: Book): Observable<any> {
```

Comment:
missing required Content-Type header. http client implicitly use json media type, but it's recommended to specify explicitly media type sent from client to server